ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -323,9 +357,166 @@ func (ipfs *Connector) pinLsHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	respBytes, _ := json.Marshal(pinLs)
+	resBytes, _ := json.Marshal(pinLs)
 	w.WriteHeader(http.StatusOK)
-	w.Write(respBytes)
+	w.Write(resBytes)
+}
+
+func (ipfs *Connector) addHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
There is benefit of pinning at the source as the source already has the content. Not only HDD is resource but bandwidth is one too.